src/spatial_reaction_systems/spatial_reactions.jl

```diff
@@ -0,0 +1,96 @@
+### Spatial Reaction Structures ###
+
+# Abstract spatial reaction structures.
+abstract type AbstractSpatialReaction end
+
+### EdgeParameter Metadata ###
+
+# Implements the edgeparameter metadata field.
+struct EdgeParameter end
+Symbolics.option_to_metadata_type(::Val{:edgeparameter}) = EdgeParameter
+
+# Implements the isedgeparameter check function.
+isedgeparameter(x::Num, args...) = isedgeparameter(Symbolics.unwrap(x), args...)
+function isedgeparameter(x, default = false)
+    p = Symbolics.getparent(x, nothing)
+    p === nothing || (x = p)
+    Symbolics.getmetadata(x, EdgeParameter, default)
+end
+
+### Transport Reaction Structures ###
+
+# A transport reaction. These are simple to hanlde, and should cover most types of spatial reactions.
+# Only permit constant rates (possibly consisting of several parameters).
+struct TransportReaction <: AbstractSpatialReaction
+    """The rate function (excluding mass action terms). Currently only constants supported"""
+    rate::Num
```

Comment:
The `rate` is not necessarily a single parameter, hence the `Num`. E.g. if you want a rate that is the product of two parameters `D` and `S` (maybe one which is spatially carrying and one which is not), then the `rate` would be `S*D` (although there is maybe still a better type to represent this?)

Reply:
ModelingToolkit convention, which Catalyst adheres to, is to never store `Num`s. They should always be unwraped at user/library interfaces like the constructor. Hence why I said to store the type of the rate as `Any`, since it will have a different type depending on what the `Num` wraps.

Reply:
Sorry, got confused by the unwrapping thing and another discussion (still not actually sure what this phrase mean, maybe you can explain next time we meet)

Reply:
fixed